File: Part2_강의자료/Ch01_큐/문제별코드/10845_큐/10845.py
import sys
input = sys.stdin.readline
from collections import deque

N = int(input())
queue = deque([])
for _ in range(N):
    command = input().split()

    if command[0] == "push":
        queue.append(int(command[1]))

    elif command[0] == "pop":
        print(queue.popleft() if queue else -1)

    elif command[0] == "size":
        print(len(queue))

    elif command[0] == "empty":
        print(0 if queue else 1)

    elif command[0] == "front":
        print(queue[0] if queue else -1)

    elif command[0] == "back":
        print(queue[-1] if queue else -1)

File: Part2_강의자료/Ch01_큐/문제별코드/10845_큐/10845.cpp
#include <stdio.h>
#include <string.h>
#include <queue>
using namespace std;

int main()
{
    int N;
    queue<int> q;
    char cmd[6];
    int value = -1;

    scanf("%d", &N);
    while (N--)
    {
        scanf("%s", cmd);
        if (strcmp(cmd, "push") == 0)
        {
            scanf("%d", &value);
            q.push(value);
        }
        else if (strcmp(cmd, "pop") == 0)
        {
            printf("%d\n", q.empty() ? -1 : q.front());
            if (!q.empty()) q.pop();
        }
        else if (strcmp(cmd, "size") == 0)
        {
            printf("%d\n", q.size());
        }
        else if (strcmp(cmd, "empty") == 0)
        {
            printf("%d\n", q.empty() ? 1 : 0);
        }
        else if (strcmp(cmd, "front") == 0)
        {
            printf("%d\n", q.empty() ? -1 : q.front());
        }
        else if (strcmp(cmd, "back") == 0)
        {
            printf("%d\n", q.empty() ? -1 : q.back());
        }
    }
    return 0;
}
